CMakeLists.txt

```diff
@@ -414,10 +417,97 @@ function(check_working_std_regex result_var_name)
+  
+  # Darwin SDK names don't match their stdlib-deployment-target names
+  if("${target_sdk}" STREQUAL "macosx")
+    set(target_sdk "OSX")
+  elseif("${target_sdk}" STREQUAL "iphoneos")
+    set(target_sdk "IOS")
+  elseif("${target_sdk}" STREQUAL "iphonesimulator")
+    set(target_sdk "IOS_SIMULATOR")
+  elseif("${target_sdk}" STREQUAL "appletvos")
+    set(target_sdk "TVOS")
+  elseif("${target_sdk}" STREQUAL "appletvsimulator")
+    set(target_sdk "TVOS_SIMULATOR")
+  elseif("${target_sdk}" STREQUAL "watchos")
+    set(target_sdk "WATCHOS")
+  elseif("${target_sdk}" STREQUAL "watchsimulator")
+    set(target_sdk "WATCHOS_SIMULATOR")
+  endif()
+
+  string(TOUPPER "${target_sdk}" target_sdk)
+  set("${result_var_name}" "${target_sdk}" PARENT_SCOPE)
+endfunction()
+
+function(get_arch_from_stdlib_target stdlib_target result_var_name)
+  string(REPLACE "-" ";" target_components "${stdlib_target}")
+  list(GET target_components 1 target_arch)
+  set("${result_var_name}" "${target_arch}" PARENT_SCOPE)
+endfunction()
+
+# Calculate the triple for an sdk/architecture.
+# Will only return a result for supported targets.
+function(get_triple sdk arch result_var_name)
+  if("${sdk}" STREQUAL "LINUX")
+    if("${arch}" STREQUAL "x86_64")
+      set(triple "x86_64-unknown-linux-gnu")
+    elseif("${arch}" STREQUAL "aarch64")
+      set(triple "aarch64-unknown-linux-gnu")
+    elseif("${arch}" MATCHES "(powerpc64|powerpc64le)")
+      set(triple "${arch}-unknown-linux-gnu")
+    elseif("${arch}" MATCHES "s390x")
+      set(triple "s390x-unknown-linux-gnu")
+    elseif("${arch}" MATCHES "(armv6|armv7)")
+      set(triple "${arch}-unknown-linux-gnueabihf")
+    endif()
+  elseif("${sdk}" STREQUAL "ANDROID")
+    if("${arch}" STREQUAL "armv7")
+      set(triple "armv7-none-linux-androideabi")
+    endif()
+  elseif("${sdk}" STREQUAL "FREEBSD")
+    if("${arch}" STREQUAL "x86_64")
+      # FIXME: Using the host OS version won't produce correct results for
+      # cross-compilation.
+      string(REPLACE "[-].*" "" FREEBSD_SYSTEM_VERSION ${CMAKE_SYSTEM_VERSION})
+      message(STATUS "FreeBSD Version: ${FREEBSD_SYSTEM_VERSION}")
+      set(triple "x86_64-unknown-freebsd${FREEBSD_SYSTEM_VERSION}")
+    endif()
+  elseif("${sdk}" STREQUAL "CYGWIN")
+    if("${arch}" STREQUAL "x86_64")
+      set(triple "x86_64-unknown-windows-cygnus")
+    endif()
+  #FIXME: Add Darwin sdks
+  endif()
+
+  set("${result_var_name}" "${triple}" PARENT_SCOPE)
+endfunction()
+
+# Set SWIFT_HOST_VARIANT_SDK/ARCH
+# - Parse from SWIFT_HOST, if given
+# - Derive from CMAKE_SYSTEM_NAME/PROCESSOR otherwise
+
+if(SWIFT_HOST)
+  get_sdk_from_stdlib_target("${SWIFT_HOST}" SWIFT_HOST_VARIANT_SDK_default)
+  get_arch_from_stdlib_target("${SWIFT_HOST}" SWIFT_HOST_VARIANT_ARCH_default)
+  if("${SWIFT_HOST_VARIANT_SDK_default}" STREQUAL "" OR
+     "${SWIFT_HOST_VARIANT_ARCH_default}" STREQUAL "")
+      message(FATAL_ERROR "Invalid SWIFT_HOST; cannot be parsed.")
+  endif()
```

Comment:
This is CMake. The build-script has changed very little (removed SWIFT_SDKs and renamed related variables). The python target calculation has changed because we shouldn't configure the host any more (we never needed to anyway, CMake always configured it, but now the configure param is only for cross-compiled stdlibs, so including the host would be an error)